packages/api-format/src/codec/base/EnumType.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017-2018 @polkadot/api-format authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { Base } from '../types';
+
+export default class CodecEnumType implements Base<Base<any>> {
```

Comment:
This is something that based on an enum, it decodes to a different type. The same comments on the enum applies, but it really should probably actually include enum instead of re-doing the logic. (composition?)